luarules/gadgets/game_team_power_watcher.lua

```diff
@@ -0,0 +1,294 @@
+		if UnitDefs[unitDefID].power then
+            unitsWithPower[unitID] = {unitID = unitID, power = UnitDefs[unitDefID].power, team = unitTeam}
+            teamPowers[unitTeam] = (teamPowers[unitTeam] + UnitDefs[unitDefID].power) or UnitDefs[unitDefID].power
+            highestTeamPower = TPW_HighestTeamPower()
+            averageTeamPower = TPW_AverageTeamPower()
+            averageHumanTeamPower = TPW_AverageHumanTeamPower(unitTeam)
+            averageAlliedTeamPower = TPW_AverageAlliedTeamPower(unitTeam)
+            averageTechGuesstimate = TPW_AverageTechGuesstimate()
+            averageAlliedTechGuesstimate = TPW_AverageAlliedTechGuesstimate(unitTeam)
+            averageHumanTechGuesstimate = TPW_AverageHumanTechGuesstimate()
+
+            --update peak powers
+            if teamPowers[unitTeam] and peakTeamPowers[unitTeam] < teamPowers[unitTeam] then
+                peakTeamPowers[unitTeam] = teamPowers[unitTeam]
+            end
+
+
+            --Debug row, eventually to be replaced with %frame triggered calculation events
+            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "highest", highestTeamPower.teamID, highestTeamPower.power)
+            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "averageTeamPower", averageTeamPower)
+            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "averageAlliedTeamPower", averageAlliedTeamPower)
+            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "averageTechGuesstimate", averageTechGuesstimate)
+            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "averageHumanTechGuesstimate", averageHumanTechGuesstimate)
+            Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "averageAlliedTechGuesstimate", averageAlliedTechGuesstimate)
+            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "peakTeamPowers", peakTeamPowers[unitTeam])
+		end
+	end
+
+
+
+	function gadget:UnitDestroyed(unitID, unitDefID, unitTeam)
+        if unitsWithPower[unitID] then
+			unitsWithPower[unitID] = nil
+		end
+        if UnitDefs[unitDefID].power then
+            if teamPowers[unitTeam] then
+                if teamPowers[unitTeam] <= UnitDefs[unitDefID].power then
+                    teamPowers[unitTeam] = 0
+                else
+                    teamPowers[unitTeam] = teamPowers[unitTeam] - UnitDefs[unitDefID].power
+                end
+            end
+            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam])
+        end
+	end
+
+
+
+    function TPW_HighestTeamPower()
+        local power = 0
+        local teamID = nil
+    
+        for t, p in pairs(teamPowers) do
```

Comment:
Please avoid single letter identifiers, even in temporary variables, their meaning must be parsed and may be parsed incorrectly.

Reply:
Done